docusaurus/docs/shared/implement-resource-handler.md

```diff
@@ -0,0 +1,189 @@
+  mux.HandleFunc("/projects", p.handleProjects)
+  mux.HandleFunc("/device", p.updateDevice)
+  p.resourceHandler := httpadapter.New(mux)
+  return p
+}
+
+func (p *MyPlugin) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+  return p.resourceHandler.CallResource(ctx, req)
+}
+
+func (p *MyPlugin) handleNamespaces(rw http.ResponseWriter, req *http.Request) {
+  if req.Method != http.MethodGet {
+    rw.WriteHeader(http.StatusNotFound)
+    return
+  }
+
+  ctxLogger := backend.Logger.FromContext(req.Context())
+
+  rw.Header().Add("Content-Type", "application/json")
+  _, err := rw.Write([]byte(`{ "namespaces": ["ns-1", "ns-2"] }`))
+  if err != nil {
+    ctxLogger.Error("Failed to write response", "error", err)
+    return
+  }
+  rw.WriteHeader(http.StatusOK)
+}
+
+func (p *MyPlugin) handleProjects(rw http.ResponseWriter, req *http.Request) {
+  if req.Method != http.MethodGet {
+    rw.WriteHeader(http.StatusNotFound)
+    return
+  }
+
+  ctxLogger := backend.Logger.FromContext(req.Context())
+
+  rw.Header().Add("Content-Type", "application/json")
+  _, err := rw.Write([]byte(`{ "projects": ["project-1", "project-2"] }`))
+  if err != nil {
+    ctxLogger.Error("Failed to write response", "error", err)
+    return
+  }
+  rw.WriteHeader(http.StatusOK)
+}
+
+func (p *MyPlugin) updateDevice(rw http.ResponseWriter, req *http.Request) {
+  if req.Method != http.MethodPost {
+    rw.WriteHeader(http.StatusNotFound)
+    return
+  }
+
+  if req.Body == nil {
+    rw.WriteHeader(http.StatusBadRequest)
+    return
+  }
+
+  ctxLogger := backend.Logger.FromContext(req.Context())
+
+  defer func() {
+    if err := req.Body.Close(); err != nil {
+      ctxLogger.Warn("Failed to close response body", "error", err)
+    }
+  }()
+
+  var payload map[string]any
+  b, err := io.ReadAll(req.Body)
+  if err != nil {
+    ctxLogger.Error("Failed to read request body to bytes", "error", err)
+    rw.WriteHeader(http.StatusInternalServerError)
+    return
+  }
+
+  err = json.Unmarshal(b, &payload)
+  if err != nil {
+    ctxLogger.Error("Failed to unmarshal request body to JSON", "error", err)
+    rw.WriteHeader(http.StatusInternalServerError)
+    return
+  }
+
+  state := payload["state"]
+  // update device with state...
+
+  rw.Header().Add("Content-Type", "application/json")
+  _, err = rw.Write([]byte(fmt.Sprintf(`{ "message": "device updated", "state": "%s" }`, state)))
+  if err != nil {
+    ctxLogger.Error("Failed to write response", "error", err)
+    return
+  }
+
+  rw.WriteHeader(http.StatusOK)
+}
+```
+
+#### Accessing the backend plugin context
+
+Use the `backend.PluginConfigFromContext` function to access `backend.PluginContext`:
+
+```go
+func (p *MyPlugin) handleNamespaces(rw http.ResponseWriter, req *http.Request) {
+	pCtx := backend.PluginConfigFromContext(req.Context())
+	ctxLogger := backend.Logger.FromContext(req.Context())
```

Comment:
it'd be nice to have an example here of how to access the plugin configuration from this handler using the context.

Reply:
I think that's a separate topic on its own so I'll create a new issue for that